packages/ui/VirtualList/VirtualListBase.js

```diff
@@ -666,6 +637,11 @@ const VirtualListBaseFactory = (type) => {
 		}
 
 		positionItems () {
+			this.prevFirstIndex = (
+				this.prevFirstIndex === this.state.firstIndex ||
+				(this.prevProps.childProps || this.props.childProps) && this.prevProps.childProps !== this.props.childProps
+			) ? -1 : this.prevFirstIndex;
```

Comment:
The code above is moved from the following code in `UNSAFE_componentWillUpdate()`.
```
UNSAFE_componentWillUpdate (nextProps, nextState) {
	if (this.state.firstIndex === nextState.firstIndex || this.props.childProps && this.props.childProps !== nextProps.childProps) {
		this.prevFirstIndex = -1; // force to re-render items
	}
}
```